src/main/java/controllers/dialogs/ConfigDialog.java

```diff
@@ -0,0 +1,166 @@
+package controllers.dialogs;
+
+import api.PSPort;
+import api.PSPortFactory;
+import com.jfoenix.controls.JFXButton;
+import com.jfoenix.controls.JFXComboBox;
+import com.jfoenix.controls.JFXTextField;
+import controllers.validators.ValidTypeValidator;
+import de.jensd.fx.glyphs.fontawesome.FontAwesomeIcon;
+import de.jensd.fx.glyphs.fontawesome.utils.FontAwesomeIconFactory;
+import javafx.event.EventType;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.scene.control.Label;
+import model.AppMain;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+
+/**
+ * Dialog class to show the application start middleware setup dialog.
+ * @author Gorka Olalde
+ */
+public class ConfigDialog extends Dialog<PSPort> {
+
+    private static final Logger LOGGER = LogManager.getLogger(ConfigDialog.class);
+
+    @FXML
+    private Label headerLabel;
+
+    @FXML
+    private JFXTextField addressField;
+
+    @FXML
+    private JFXTextField portField;
+
+    @FXML
+    private JFXComboBox<String> typeCombo;
+
+    @FXML
+    private ValidTypeValidator addressValidator;
+
+    @FXML
+    private ValidTypeValidator portValidator;
+
+    private JFXButton saveBtn;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Perhaps 'saveBtn' could be replaced by a local variable.](https://www.codacy.com/app/POPBL6/front-end/file/2720643955/issues/source?bid=3316499&fileBranchId=3347636#l49)